R/DiseasySeason.R

```diff
@@ -0,0 +1,504 @@
+                                                        "Constant (no) seasonality model.",
+                                                        "Risk of infection constant through year")
+
+      attr(private$.model_t, "name")         <- attr(private$.model_date, "name")
+      attr(private$.model_t, "description")  <- attr(private$.model_date, "description")
+
+      # Logging
+      private$lg$info("Using constant_season model")
+    },
+
+
+    #' @description
+    #'   Sets the `DiseasySeason` module to use a cosine model for season.
+    #' @param peak (`numeric`)\cr
+    #'   Sets the period of maximal activity (days past new-year).
+    #'   By default, risk of infection is antiphase with the DMI climate normal of the maximum daily temperature.
+    #' @param scale `r rd_scale()`
+    use_cosine_season = function(peak = 20.09946, scale = 0.5726693) {
+
+      # Check parameters
+      coll <- checkmate::makeAssertCollection()
+      if (is.null(self$reference_date)) coll$push(private$reference_date_not_set_error())
+      checkmate::assert_number(peak,  lower = 0, upper = 365, add = coll)
+      checkmate::assert_number(scale, lower = 0, upper = 1,   add = coll)
+      checkmate::reportAssertions(coll)
+
+      # Create normalizer
+      date_map <- \(date) lubridate::decimal_date(date) %% 1
+
+      # Create the main season function
+      cosine_season <- \(t) 1 - scale * (1 - cos(2 * pi * (t - peak / 365))) / 2
+
+      # Determine the references
+      reference_t     <- date_map(self$reference_date)
+      reference_value <- cosine_season(reference_t)
+
+      # Set the models
+      private$.model_date <- \(date) cosine_season(date_map(date)) / reference_value
+      private$.model_t    <- \(t)    cosine_season(reference_t + t / 365) / reference_value
+
+      attr(private$.model_date, "name")        <- "cosine_season"
+      attr(private$.model_date, "description") <- paste(sep = "\n",
+                                                        "Simple seasonality model.",
+                                                        "Risk of infection highest at the `peak` days after new year")
+      attr(private$.model_date, "dots")       <- list(scale = scale)
+
+      attr(private$.model_t, "name")         <- attr(private$.model_date, "name")
+      attr(private$.model_t, "description")  <- attr(private$.model_date, "description")
+      attr(private$.model_t, "dots")         <- attr(private$.model_date, "dots")
+
+      # Logging
+      private$lg$info("Using cosine_season model")
+    },
+
+
+    #' @description
+    #'   Sets the `DiseasySeason` module to use the first version of the covid 19 season model
+    #' @param scale `r rd_scale()`
+    use_covid_season_v1 = function(scale = 0.4825524) {
+
+      # Determine what the max_scale can be
+      # For that we need the parameters of the season model fit
+      b <- 0.12298820
+      k <- 0.07675696
+      t0 <- 10.53276565
```

Comment:
It is not apparent to me what `b`, `k` and `t0` are. Are they common modelling terms?